elooplib/client.cpp

```diff
@@ -6,22 +6,22 @@ Client::Client(Client &&rhs)
       request_field(std::move(rhs.request_field)),
       response(std::move(rhs.response)) {}
 
-int Client::handleConnection()
+std::shared_ptr<Node> Client::handleConnection()
 {
   char buff[512];
-  size_t rcv;
+  int rcv;
 
   rcv = recv(getFd(), buff, 512, 0);
   if (rcv < 0)
   {
-    return -1;
+    return nullptr;
```

Comment:
Do you know what constructor will be called here? Will it work if ctor will be `explicit`?